api/all/src/main/java/io/opentelemetry/api/logs/LogRecord.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.logs;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.context.Context;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Used to construct and emit logs from a {@link Logger}.
+ *
+ * <p>Obtain a {@link LogRecordBuilder} via {@link Logger#eventBuilder(String)}, add properties using the
+ * setters, and emit the event by calling {@link #emit()}.
+ */
+public interface LogRecord {
+
+  /** Set the epoch timestamp using the timestamp and unit.
+   *
+   * <p>This is the time when the event occured.
+   * */
+  LogRecord setEpoch(long timestamp, TimeUnit unit);
+
+  /** Set the epoch timestamp using the instant.
+   *
+   * <p>This is the time when the event occured.
+   * */
+  LogRecord setEpoch(Instant instant);
+
+  /** Set the context. */
+  LogRecord setContext(Context context);
+
+  /** Set the severity. */
+  LogRecord setSeverity(Severity severity);
+
+  /** Set the severity text. */
+  LogRecord setSeverityText(String severityText);
+
+  /** Set the body string. */
+  LogRecord setBody(String body);
+
+  /** Set the attributes.
+   *
+   * <p>Merges with the attributes already present
+   * */
+  LogRecord setAttributes(Attributes attributes);
+
+  /** Mark the completion of building LogRecord.
+   *
+   * <p>Only the timing of the first emit call for a given {@code LogRecord} will be recorded in the
+   * <i>observed timestamp</i>, and implementations are free to ignore all further calls.
+   * */
+  void emit();
+
+  /** Set the LogRecord as being an event using name as the event name, along with the attributes.
+   *
+   * <p>This method is equivalent to Span.addEvent and provides a path for moving to using LogRecord
+   * for Span Events as against Span.Event. The event name itself is set as another attribute with
+   * the semantic convention of "otel.event.name" as the attribute key.
```

Comment:
Can we use "event.name" instead of "otel.event.name"?

I think we can have the "event.*" namespace all dedicated to events and I think "otel." prefix is not necessary.

Reply:
Yeah, we could do that. We need to define the expected behaviour if the attribute set passed as event data has an attribute with the same name `event.name`.  I will go with dropping that attribute and use only the one explicitly passed in the API call parameter.